### PR TITLE
feat: generate contact bundle and populate proto

### PR DESCRIPTION
**Pull request by michaelx11**
## Overview

This can wait until the project restructure PR #87 , but provides an initial example of using new protos with vmac. This uses @richardhuaaa 's contact bundle proto structure from the MudPup sprint.

Notably - `tonic` does not build for `wasm`, so I needed to gate `tonic` behind a feature "grpc" inside xmtp-proto. xmtp-networking then relies on that feature. FYI @neekolas so that doesn't catch you by surprise.
## Test Plan

Unit tests pass

### CODE REVIEW

File: xmtpv3/crates/libxmtp-core/src/vmac.rs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+) -> VmacUnsignedPublicKey {
+    VmacUnsignedPublicKey {
+        created_ns: 0,
+        union: Some(Union::Curve25519(VodozemacCurve25519 {
+            bytes: key.to_bytes().to_vec(),
+        })),
+    }
+}
+
+fn account_linked_key_from_curve25519_public_key(key: Curve25519PublicKey) -> VmacAccountLinkedKey {
```
Comment (neekolas) [inline xmtpv3/crates/libxmtp-core/src/vmac.rs:27]: Wouldn't this be more idiomatic as a `impl from<Curve25519PublicKey> for VmacAccountLinkedKey {`

Comment (michaelx11) [reply xmtpv3/crates/libxmtp-core/src/vmac.rs:27]: I might be misinterpreting, but Rust's orphan rules don't allow me to write an Impl where both the trait and type live in different crates. Now that I think about it we maybe can work around it with either a Wrapper or a fake Generic type?

https://stackoverflow.com/q/75765502

Comment (michaelx11) [reply xmtpv3/crates/libxmtp-core/src/vmac.rs:27]: This is what the compiler spits at me:

```
error[E0117]: only traits defined in the current crate can be implemented for types defined outside of the crate
  --> crates/libxmtp-core/src/vmac.rs:27:1
   |
27 | impl From<Curve25519PublicKey> for VmacUnsignedPublicKey {
   | ^^^^^-------------------------^^^^^---------------------
   | |    |                             |
   | |    |                             `VmacUnsignedPublicKey` is not defined in the current crate
   | |    `Curve25519PublicKey` is not defined in the current crate
   | impl doesn't use only types from inside the current crate
   |
   = note: define and implement a trait or new type instead
```

Comment (neekolas) [reply xmtpv3/crates/libxmtp-core/src/vmac.rs:27]: I also totally skimmed past your comment above the first function here

Comment (michaelx11) [reply xmtpv3/crates/libxmtp-core/src/vmac.rs:27]: Very understandable it's a tiny comment punting a large problem haha

Comment (neekolas) [reply xmtpv3/crates/libxmtp-core/src/vmac.rs:27]: Ya. A type wrapper would work, but I'm not sure of the downstream consequences of actually using the wrapped type.

**Pull request closed without merging**